srtcore/channel.h

```diff
@@ -255,7 +248,7 @@ class CChannel
     // calling ::sendmsg function. It uses a static buffer to supply data
     // for the call, and it's stated that only one thread is trying to
     // use a CChannel object in sending mode.
-    bool setSourceAddress(msghdr& mh, const sockaddr_any& adr) const
+    bool setSourceAddress(msghdr& mh, void *buf, const sockaddr_any& adr) const
```

Comment:
If the variable passed here is a local variable created in the calling function, isn't it simpler to simply create this buffer inside this very function?

BTW. `void*` is "transparent" in C++, so use the original type and do not cast.

Reply:
That would not work. Inside this function, mh.msg_control is set to buf. If this was declared in this scope, it would be destroyed when exiting the function, and the information set to the cmsghdr would be lost. 
I thought about moving `mh.msg_control = (void *) buf;` to outside of setSourceAddress, but that didn't feel right

Reply:
Ah, right, missed that. So yes, it should then stay as is, but the type of the `buf` parameter should be `char*`, as the original one.